enhancements/authentication/adding-uid-extra-claim-mappings-to-oidc.md

```diff
@@ -0,0 +1,417 @@
+kind: Authentication
+metadata:
+  name: cluster
+spec:
+  type: OIDC
+  oidcProviders:
+  - name: foo
+    claimMappings:
+      uid:
+       expression: 'claims.email'
+```
+
+To configure the UID of a cluster user identity using a specific claim value on HyperShift, a Cluster Administrator creates a `HostedCluster` resource to populate the claim mapping like so:
+```yaml
+apiVersion: hypershift.openshift.io/v1alpha1
+kind: HostedCluster
+metadata:
+  name: example
+  namespace: clusters
+spec:
+  configuration:
+    authentication:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            claim: email
+```
+
+To configure the UID of a cluster user identity using a CEL expression on HyperShift, a Cluster Administrator creates a `HostedCluster` resource to populate the claim mapping like so:
+
+```yaml
+apiVersion: hypershift.openshift.io/v1alpha1
+kind: HostedCluster
+metadata:
+  name: example
+  namespace: clusters
+spec:
+  configuration:
+    authentication:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            expression: claims.email
+```
+
+##### Scenario 2: Configuring the Extra values of cluster user identities
+An OpenShift/HyperShift customer is using an external Identity Provider to manage user authentication in their organization. A Cluster Administrator
+would like to use this existing authentication layer for authenticating users to their clusters. They have already configured this using the existing
+external OIDC workflows, but would like the cluster user identities to include Extra values based on custom claims they have configured their
+Identity Provider to include in the issued JWT authentication tokens. Specifically, they are using the `team` and `role` claims. They are using a
+third-party authorizer to implement access control guardrails based on the combination of the team a user is part of and the role they are given.
+
+To configure the Extra values of a cluster user identity value on OpenShift, a Cluster Administrator updates the `authentications.config.openshift.io/cluster` resource
+to populate the claim mapping like so:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+  name: cluster
+spec:
+  type: OIDC
+  oidcProviders:
+  - name: foo
+    claimMappings:
+      extra:
+      - key: 'myorg.io/team'
+        valueExpression: claims.team
+      - key: 'myorg.io/role'
+        valueExpression: claims.role
+```
+
+To configure the Extra values of a cluster user identity on HyperShift, a Cluster Administrator creates a `HostedCluster` resource to populate the claim mapping like so:
+```yaml
+apiVersion: hypershift.openshift.io/v1alpha1
+kind: HostedCluster
+metadata:
+  name: example
+  namespace: clusters
+spec:
+  configuration:
+    authentication:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          extra:
```

Comment:
can I create one these that says I'm a node? Should I be able to?

Can I create a configuration that makes fake service account tokens?  Should I be able to?

Reply:
We don't allow the use of the reserved `kubernetes.io` domain or `*.kubernetes.io` subdomains. So from the `extra` attribute specifically, AFAIK this cannot be done.

Reply:
what about openshift scopes?

Reply:
Also won't allow the `openshift.io` domain or `*.openshift.io` domains.

If there are additional domains we should mark as reserved, let me know.